src/image/roi/manager.js

```diff
@@ -301,10 +301,47 @@ export default class RoiManager {
         return this;
     }
 
+
+    appendRelated(roiMap) {
+        let allRois = this.getRois();
+        let allRelated = {related: []};
+        for (let i = 0; i < allRois.length; i++) {
+            let x = allRois[i].minX;
+            let y = allRois[i].minY;
+            let allPoints = allRois[i].points;
+            let currentRelated = correspondingRoisAndPixels(x, y, allPoints, roiMap);
+            allRelated.related.push(currentRelated);
+        }
+        return allRelated;
+    }
+
+
     _assertLayerWithLabel(label) {
         if (!this._layers[label]) {
             throw new Error(`no layer with label ${label}`);
         }
     }
+
+
 }
 
+function correspondingRoisAndPixels(x, y, points, roiMap) {
+    let relatedRois = {id: [], pixels: []};
```

Comment:
For performance if you know that your array has an upper size limit you can create an array with this size and then modify his length attribute, something like this

```js
let arr = new Array(7);
for (var i = 0; i < 5; i++) {
  arr[i] = i;
}
arr.length = 5;
```